docs/dev/query-automatic-acceleration.md

```diff
@@ -0,0 +1,75 @@
+In a database engine, there are different ways to optimize query performance. For instance, rule-based/cost-based optimizer and distributed execution layer tries to find best execution plan by cost estimate and equivalent transformation of query plan. Here we're proposing an alternative approach which is to accelerate query execution by materialized view for time-space tradeoff.
+
+### Architecture
+
+Here is a reference architecture that illustrates components and the entire workflow which essentially is a *workload-driven feedback loop*:
+
+1. *Input*: Query plan telemetry collected
+2. *Generating feedback*: Feeding it into a workload-driven feedback generator
+3. *Output*: Feedback for optimizer to rewrite query plan in future
+
+Basically, *feedback* is referring to various materialized view prebuilt (either online or offline) which hints acceleration opportunity to query optimizer.
+
+![AutoMV (1) (1)](https://user-images.githubusercontent.com/46505291/168863085-d5d39ab4-40bf-41c8-922c-4a45572e40dd.png)
+
+There are 2 areas and paths moving forward for both of which lack open source solutions:
+
+* OpenSearch engine acceleration: accelerate DSL or SQL/PPL engine execution
+* MPP/Data Lake engine acceleration: accelerate Spark, Presto, Trino
```

Comment:
References to other DB are unclear. Are you comparing OpenSearch's approach with theirs?